src/communication/websocket/auth.cpp

```diff
@@ -16,4 +16,11 @@ namespace communication::websocket {
 bool SafeAuth::Authenticate(const std::string &username, const std::string &password) const {
   return auth_->Lock()->Authenticate(username, password).has_value();
 }
+
+bool SafeAuth::DoesUserHasPermission(const std::string &username, const auth::Permission permission) const {
+  if (const auto user = auth_->Lock()->GetUser(username); user) {
+    return user->GetPermissions().Has(permission) == auth::PermissionLevel::GRANT;
+  }
+  return false;
```

Comment:
What if memgraph doesn't have any users? Shouldn't we allow then everything similarly to `AuthChecker::IsUserAuthorized` in `memgraph.cpp`? I think it would the best to push down that AuthChecker, because that is meant to do this exactly: check if a user has permission or not.

Reply:
Yes that makes sense, to only authenticate/authorize if there are any users

Reply:
@jbajic pointed out that `AuthChecker` works with privileges from the query engine point of view, and not with permissions. I agree with him that the two shouldn't be mixed, so for now it is okay to go with a different solution.